### DISCUSSION TITLE
Fix for "Version 2.6.x not found" build errors

### DISCUSSION THREAD

**Discussion by brockfanning** [open-sdg/open-sdg#1884] [Announcements]
Github has recently stopped maintaining their "action" for Ruby, which many Open SDG sites may be using. If you are affected by this issue, your site builds will be failing with an error message like this:

```
Run actions/setup-ruby@v1
  with:
    ruby-version: 2.6.x
------------------------
NOTE: This action is deprecated and is no longer maintained.
Please, migrate to https://github.com/ruby/setup-ruby, which is being actively maintained.
------------------------
Error: Version 2.6.x not found
```

To fix this error, you will need to edit the files in your site repository's `.github/workflows` folder. In each of the three files there you will need to make the following changes:

* Look for `uses: actions/setup-ruby@v1` and change it to `uses: ruby/setup-ruby@v1`
* Look for `ruby-version: 2.6.x` and change it to `ruby-version: '2.6'`

Here is an example of the fix from our "site starter" repository:
* https://github.com/open-sdg/open-sdg-site-starter/commit/739842339d4e98936fc6b9e699d5d929a5df006c

Note this needs to be done for each of the three files:

![image](https://user-images.githubusercontent.com/1319083/206204401-40189825-8e19-4e9e-a888-16fd7241901b.png)

**Comment by christoph-LSN** [comment]
@brockfanning perhaps it's worth to mention that also the deploy-to-production.yml has to be changed as well

**Reply by brockfanning** [reply]
Good point, I've updated the post.

**Reply by sarah-beasley** [reply]
Please note that any other file in the .github/workflows folder that mentions ruby-version: 2.6.x, needs to be updated as above too. For example, the UK site has a deploy-feature-branch.yml in the .github/workflows folder that also needed to be updated in order for builds to complete.

**Comment by city-of-cape-town** [comment]
Hi @brockfanning

I tried to update the workflows as instructed above but All checks have failed.

Would you please assist. This has been persistent on both repositories [site](https://github.com/city-of-cape-town/site-2-0-0/actions) and [data](https://github.com/city-of-cape-town/data-2-0-0/actions) over the past few days.

Thank you in advance.

**Reply by sarah-beasley** [reply]
Have you created one feature branch where you've made the changes to all three files within the same feature branch? Once all three are done, it should build successfully and then you can merge that one feature branch to develop. Hope this helps!

**Reply by MoBosse** [reply]
If I am not mistaken you still have a `2.6.x` in your test-pull-request file here:
https://github.com/city-of-cape-town/site-2-0-0/blob/develop/.github/workflows/test-pull-requests.yml#L15

**Reply by city-of-cape-town** [reply]
@sarah-beasley and @MoBosse Thank you so much, both approaches worked. 

I had to make changes to all three files within the same feature branch as @sarah-beasley suggested and it successfully merged with the develop branch. Also, I noted that the updating of each file had to follow a specific order, after a couple of attempts with randomized updating, all checks passed only after updating test-pull-request workflow first as @MoBosse recommended.

Thanks again guys.
